clusterloader2/framework/framework.go

```diff
@@ -0,0 +1,108 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package framework
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/perf-tests/clusterloader2/api"
+	"k8s.io/perf-tests/clusterloader2/framework/util"
+)
+
+const (
+	automanagedNamespaceName = "Namespace"
+)
+
+func convertObjectTypeToGVK(objectType *api.ObjectType) schema.GroupVersionKind {
+	return schema.GroupVersionKind{
+		Group:   objectType.APIGroup,
+		Version: objectType.APIVersion,
+		Kind:    objectType.Kind,
+	}
+}
+
+// Framework allows to execute cluster related actions
```

Comment:
// Framework allows for interacting with Kubernetes cluster via
// official Kubernetes client.

or sth like that
- you don't perform any actions - you just send requests changing the desired state
- [also namespace is also an objects, it's not special concept]